src/main/java/com/fasterxml/jackson/dataformat/xml/util/DefaultXmlPrettyPrinter.java

```diff
@@ -126,6 +133,30 @@ public void indentObjectsWith(Indenter i)
 
     public void spacesInObjectEntries(boolean b) { _spacesInObjectEntries = b; }
 
+    /**
+     * Initialize lineFeed with systemDefault.
+     * @since 2.15
+     */
+    static {
+        String lf = null;
+        try {
+            lf = System.getProperty("line.separator");
+        } catch (Throwable t) { } // access exception?
+        _lineFeed = lf;
+    }
+
+    /**
+     * Sets custom linefeed
+     * @since 2.15
+     */
+    public DefaultXmlPrettyPrinter withCustomlineFeed(String lineFeed) {
+        // 06-Feb-2023, joohyukkim: when JacksonException extends RuntimeExceptions, throw it?
+        if (lineFeed != null) {
+            _lineFeed = lineFeed;
+        }
+        return this;
```

Comment:
a non-static method updating a static field - I think this is not good practice

Reply:
I would suggest that you should have a static DEFAULT_LINE_FEED variable set in the static initializer. (non-sttaic) _lineFeed is initialised equal to DEFAULT_LINE_FEED but `withCustomLineFeed` can modify it.

Reply:
Makes sense. I will get on it.
Thanks!

Reply:
@pjfanning I have a question.
Even if we make `_lineFeed` field non-static, we have static line-feeding Indenters like `Lf2SpacesIndenter`.
They can't access non-static field of outer class. 🥲